pkg/cmd/server/api/validation/oauth.go

```diff
@@ -40,64 +41,117 @@ func ValidateOAuthConfig(config *api.OAuthConfig) fielderrors.ValidationErrorLis
+			validationResults.AddErrors(ValidateRemoteConnectionInfo(provider.RemoteConnectionInfo).Prefix("provider")...)
 
 		case (*api.HTPasswdPasswordIdentityProvider):
-			allErrs = append(allErrs, ValidateFile(provider.File, "provider.file")...)
+			validationResults.AddErrors(ValidateFile(provider.File, "provider.file")...)
+
+		case (*api.LDAPPasswordIdentityProvider):
+			validationResults.Append(ValidateLDAPIdentityProvider(provider, identityProvider))
 
 		case (*api.GitHubIdentityProvider):
-			allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
+			validationResults.AddErrors(ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
 
 		case (*api.GoogleIdentityProvider):
-			allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
+			validationResults.AddErrors(ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
 
 		case (*api.OpenIDIdentityProvider):
-			allErrs = append(allErrs, ValidateOpenIDIdentityProvider(provider, identityProvider)...)
+			validationResults.AddErrors(ValidateOpenIDIdentityProvider(provider, identityProvider)...)
 
 		}
 	}
 
-	return allErrs
+	return validationResults
+}
+
+func ValidateLDAPIdentityProvider(provider *api.LDAPPasswordIdentityProvider, identityProvider api.IdentityProvider) ValidationResults {
+	validationResults := ValidationResults{}
+
+	if len(provider.URL) == 0 {
+		validationResults.AddErrors(fielderrors.NewFieldRequired("provider.url"))
+		return validationResults
+	}
+
+	u, err := ldappassword.ParseURL(provider.URL)
+	if err != nil {
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("provider.url", provider.URL, err.Error()))
+		return validationResults
+	}
+
+	// Make sure bindDN and bindPassword are both set, or both unset
+	// Both unset means an anonymous bind is used for search (https://tools.ietf.org/html/rfc4513#section-5.1.1)
+	// Both set means the name/password simple bind is used for search (https://tools.ietf.org/html/rfc4513#section-5.1.3)
+	if len(provider.BindDN) > 0 && len(provider.BindPassword) == 0 {
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("provider.bindDN", provider.BindDN, "Cannot specify bindDN without bindPassword"))
+	}
+	if len(provider.BindPassword) > 0 && len(provider.BindDN) == 0 {
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("provider.bindPassword", "<masked>", "Cannot specify bindPassword without bindDN"))
+	}
+
+	if provider.Insecure {
+		if u.Scheme == ldappassword.ProtocolLDAPS {
+			validationResults.AddErrors(fielderrors.NewFieldInvalid("provider.url", provider.URL, fmt.Sprintf("Cannot use %s scheme with insecure=true", u.Scheme)))
+		}
+		if len(provider.CA) > 0 {
```

Comment:
I probably should have noticed above, but if I'm upgrading from insecure to secure, I don't get to check against a CA?